src/wifi/model/periodic-dmg-wifi-scheduler.cc

```diff
@@ -335,79 +293,97 @@ PeriodicDmgWifiScheduler::GetAvailableBlocks (uint32_t allocDuration, uint32_t s
 }
 
 void
-PeriodicDmgWifiScheduler::UpdateAvailableSlots (uint32_t startAlloc, uint32_t endAlloc, uint32_t difference)
+PeriodicDmgWifiScheduler::UpdateAvailableSlots (uint32_t startAlloc, uint32_t endAlloc)
 {
-  NS_LOG_FUNCTION (this);
+  NS_LOG_FUNCTION (this << startAlloc << endAlloc);
 
-  std::vector<std::pair<uint32_t, uint32_t> > newDti;
+  uint32_t startSlot;
+  uint32_t endSlot;
 
-  if (difference > 0)
+  for (auto it = m_availableSlots.begin(); it != m_availableSlots.end(); ++it)
```

Comment:
advancing the iterator in the for loop while inserting/erasing can lead to unexpected results. it's safer do to it manually (yes, you could use a while loop :) )
```suggestion
  for (auto& it = m_availableSlots.begin(); it != m_availableSlots.end();)
```

Reply:
I specifically did it in this way because I know what I am modifying and how the iterator would change according to any insertion/deletion. Also, remember that once the algorithm finds the slot that needs to be changed, it proceeds with the change and then stops. 
I would stick with my solution, with some of the improvements that you suggested (such as the referenced iterator).